tests/virt_autotest/login_console.pm

```diff
@@ -64,8 +64,10 @@ sub check_kvm_modules {
 
     # for modular libvirt, virtqemud is expected in "loaded: active or inactive" status.
     # virtqemud.socket seems to be always in "loaded: active" status
-    unless (is_monolithic_libvirtd) {
-        die 'virtqemud.socket is not running!' unless script_run("systemctl is-active virtqemud.socket") eq 0;
+    unless (get_var('TEST_SUITE_NAME') =~ /kubevirt-tests/) {
```

Comment:
This issue is specific to kubevirt-tests ?

Reply:
Kubevirt tests should ignore the libvirt checking, its tests do not depends on libvirt.

Reply:
I do not think the code orgnization here is good. You should skip your checking inside instead of outside, for example:
```
unless (is_monolithic_libvirtd) {
  unless (get_var('TEST_SUITE_NAME') =~ /kubevirt-tests/
}
```
Code block here should be grouped by checkings instead of tests.

Reply:
if so, his kubevirt test will get a failed result finally.

Reply:
He can switch the testing logic order. Put test name checking inside to skip.

Reply:
If the code logic  goes into the function "is_monolithic_libvirtd", kubevirt tests will fail because it doesn't install any libvirt packages. That's why I put it outside of the judgment of function "is_monolithic_libvirtd".
```
sub is_monolithic_libvirtd {
    record_info('WARNING', 'Libvirt package is not installed', result => 'fail') if (script_run('rpm -q libvirt-libs'));
```

Reply:
Why not install kvm pattern together during host installation phase ? You can just set PATTERNS=default,kvm. It is easy and no need to modify code at all.

Reply:
@waynechen55  JFYI. For kubevirt test, it requires the host to have kvm module installed and loaded, but it does not need qemu/virtualization stake, because they will be given in the kubevirt containers. Does that help for reviewing the PR? 
If you still have further comments, feel free to raise. I suppose @nanzhg  will follow and  reply. Otherwise, maybe we can consider merging?

Reply:
```
unless (is_monolithic_libvirtd) {
    unless (get_var('TEST_SUITE_NAME') =~ /kubevirt-tests/) {
        die 'virtqemud.socket is not running!' unless script_run("systemctl is-active virtqemud.socket") eq 0;
    }
}
```
This will still fail you kubevirt test ? @nanzhg 
I just think checking here should be grouped by types of checkings instead of test names.

Reply:
> Why not install kvm pattern together during host installation phase ? You can just set PATTERNS=default,kvm. It is easy and no need to modify code at all. The functionality already implemented which is ready to be used instantly.

I tried with the parameter PATTERNS=default,kvm, looks like it didn't take effect.
https://openqa.suse.de/tests/11768689#step/login_console/14

Reply:
> > Why not install kvm pattern together during host installation phase ? You can just set PATTERNS=default,kvm. It is easy and no need to modify code at all. The functionality already implemented which is ready to be used instantly.
> 
> I tried with the parameter PATTERNS=default,kvm, looks like it didn't take effect. https://openqa.suse.de/tests/11768689#step/login_console/14

This will work ?
```
unless (is_monolithic_libvirtd) {
    unless (get_var('TEST_SUITE_NAME') =~ /kubevirt-tests/ or script_run("systemctl is-active virtqemud.socket") eq 0) {
        die 'virtqemud.socket is not running!';
    }
}
```
Better ?

Reply:
> This will still fail you kubevirt test ? @nanzhg is_monolithic_libvirtd will return 0 or 1 and does not have assertion. I just think code here should be grouped by types of checkings instead of test names.

Yes, is_monolithic_libvirtd will return 0. When it begins to execute "is_monolithic_libvirtd", kubevirt tests will be failed as the WARNING statement will set the whole test as failed.

Reply:
> > This will still fail you kubevirt test ? @nanzhg is_monolithic_libvirtd will return 0 or 1 and does not have assertion. I just think code here should be grouped by types of checkings instead of test names.
> 
> Yes, is_monolithic_libvirtd will return 0. When it begins to execute "is_monolithic_libvirtd", kubevirt tests will be failed as the WARNING statement will set the whole test as failed.

I do no think we should fail inside a ```is_xxx``` function. ```is_xxx``` should return true or false to let test decide what to do next. You can help make it better, but not let it fail. I do no think it is very good.

Reply:
> This will work ?
> 
> ```
> unless (is_monolithic_libvirtd) {
>     unless (get_var('TEST_SUITE_NAME') =~ /kubevirt-tests/ or script_run("systemctl is-active virtqemud.socket") eq 0) {
>         die 'virtqemud.socket is not running!';
>     }
> }
> ```
> 
> Better ?

The first statement will fail the whole test on sle-micro. E.g. https://openqa.suse.de/tests/11758005#step/login_console/11

`unless (is_monolithic_libvirtd) {
`

Reply:
@waynechen55 Probably I was wrong, the record_info statement will not fail the test after double checking with a whole test (http://10.67.129.96/tests/2492). I also updated the logic of check_kvm_modules function as you suggested.

Reply:
> @waynechen55 Sorry I was wrong. After double checking with a whole test (http://10.67.129.96/tests/2492#step/login_console/11), the WARNING statement will not fail the test, it just marked info red. I also updated the logic of check_kvm_modules function as you suggested.

Looks better now